### PR TITLE
solve framed integration with actix-http

### PR DESCRIPTION
**Pull request by robjtede**
## PR Type
Bug Fix / Refactor

## PR Checklist
Check your PR fulfills the following:

- [ ] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview
- Follow up #138 and #110.
- Primarily, changes made to `Framed` while attempting integration with `actix-http`.
- Renames some methods to better describe their use.
- Adds a method to get a pinned reference to the framed IO.
- Fix things that were setting off my spell checker.

## Breaking Changes
- Many. But a major bump was already necessary for codec and utils from #138.

Quick note: will address checklist as we go but this should probably be considered in tandem with actix/actix-web#1634

### CODE REVIEW

**Comment by robjtede** [comment]
Not entirely sure why the 1.39 build is failing. There does appear to be a 1.40 feature in the crate mentioned, they do claim MSRV of 1.39 but their CI is passing?

**Comment by JohnTitor** [comment]
Because their MSRV check doesn't work correctly, submitted a fix: tokio-rs/tracing/pull/934
So, we also should bump MSRV to 1.40.0 (anyway it's fine for me since actix frends are going to 1.42.0 :) ).

**Comment by JohnTitor** [comment]
Btw, almost changes look great to me, I'll take a more look later about the `Framed` change.

File: actix-utils/CHANGES.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 * Upgrade `tokio-util` to `0.3`.
 * Remove unsound custom Cell and use `std::cell::RefCell` instead, as well as `actix-service`.
+* Provide correctly spelled `LocalWaker::is_registered` method and deprecate the replaced.
```
Comment (JohnTitor) [inline actix-utils/CHANGES.md:7]: "deprecate" here is not correct for me, I presume it's an oversight in e5b1aebc50d4fd185a71895b450beefbecb7d907.

**Review by JohnTitor — APPROVED** [review]
Overall looks good! Left one nit.

File: actix-utils/CHANGES.md (hunk 1 above)
Comment (robjtede) [reply actix-utils/CHANGES.md:7]: ahh good spot, yeah i had the change in to keep the old version but realised that we had breaking changes anyway

**Pull request merged**